File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int res = 0, temp = 0, last = 0;
        set<char> t = {};

        for(int i = 0; i < s.size(); i++) {
            for(int j = i; j < s.size(); j++) {
                if(t.find(s[j]) == t.end()) {
                    t.insert(s[j]);
                } else {
                    temp = t.size();
                    res = max(res, temp);
                    t.clear();
                    break;
                }
            }
            temp = t.size();

            res = max(res, temp);
            t.clear();
        }
        temp = t.size();

        return max(res, temp);
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res = 0
        se = set()

        l = 0
        
        for r in range(len(s)):
            while s[r] in se:
                se.remove(s[l])
                l += 1
            se.add(s[r])
            res = max(res, r - l + 1)
        
        return res